config/types/files.go

```diff
@@ -129,6 +130,19 @@ func (fc FileContents) Validate() report.Report {
 	return report.Report{}
 }
 
+func makeDataUrl(contents []byte) string {
+	opaque := "," + dataurl.Escape(contents)
+	b64 := ";base64," + base64.StdEncoding.EncodeToString(contents)
+	if len(b64) < len(opaque) {
+		opaque = b64
+	}
+	uri := (&url.URL{
+		Scheme: "data",
+		Opaque: opaque,
+	}).String()
+	return uri
+}
```

Comment:
I think formatting can be more compact:
```suggestion
func makeDataUrl(contents []byte) string {
  opaque := "," + dataurl.Escape(contents)

  if b64 := ";base64," + base64.StdEncoding.EncodeToString(contents); len(b64) < len(opaque) {
    opaque = b64
  }

  return (&url.URL{
    Scheme: "data",
    Opaque: opaque,
  }).String()
}
```

`b64` declaration can be moved outside of `if` block though, no preferences here.